modules/gateways/blockonomics/blockonomics.php

```diff
@@ -281,6 +281,28 @@ public function convertFiatToBlockonomicsCurrency($fiat_amount, $currency, $bloc
         return intval(1.0e8 * $fiat_amount / $price);
     }
 
+    /**
+     * Convert received amount based in client currency
+     *
+     * @param array $order
+     * @param string $paymentAmount
+     * @return float converted value
+     */
+    public function convertAmountToOrderCurrency($order, $paymentAmount)
+    {
+        $clientByorder = Capsule::table('tblclients')
+            ->join('tblinvoices', 'tblclients.id', '=', 'tblinvoices.userid')
+            ->join('tblcurrencies', 'tblclients.currency', '=', 'tblcurrencies.id')
+            ->where('tblinvoices.id', $order['order_id'])
+            ->first();
+
+        if ($clientByorder->code == $order['order_currency']) {
+            return floatval($paymentAmount); // no convertion is requiered
+        }
+
+        return $this->getUnderpaymentSlack(floatval($paymentAmount) * floatval($clientByorder->rate), 2);
```

Comment:
`$this->getUnderpaymentSlack` should be `$this->roundUp`.

Reply:
roundUp was removed as requested by @shivaenigma